pkg/platform/api/buildplanner/model/buildplan.go

```diff
@@ -215,59 +189,93 @@ func (b *BuildPlanByProject) CommitID() (strfmt.UUID, error) {
 	}
 
-	if b.Commit.Error != nil {
-		if b.Commit.Error.Message != "" {
-			return nil, errs.New("BuildPlanByCommit.Build: Could not get build via commit ID, API returned commit error message: %s", b.Commit.Message)
-		}
-		return nil, errs.New("BuildPlanByCommit.Build: Could not retrieve commit")
-	}
-
-	if b.Commit.Type == NotFound {
-		return nil, locale.NewError("err_buildplanner_commit_not_found", "Build plan does not contain commit")
+	if IsErrorResponse(b.Commit.Type) {
+		return nil, ProcessCommitError(b.Commit, "Could not get build from commit response")
 	}
 
 	if b.Commit.Build == nil {
-		if b.Commit.Error != nil {
-			return nil, errs.New("BuildPlanByCommit.Build: Commit not found: %s", b.Commit.Error.Message)
-		}
 		return nil, errs.New("BuildPlanByCommit.Build: Commit does not contain build")
 	}
 
-	if b.Commit.Build.PlanningError != nil {
+	if IsErrorResponse(b.Commit.Build.Type) {
+		return nil, ProcessBuildError(b.Commit.Build, "Could not get build from commit response")
+	}
+
+	return b.Commit.Build, nil
+}
+
+func (b *BuildPlanByCommit) CommitID() (strfmt.UUID, error) {
+	if b.Commit == nil {
+		return "", errs.New("BuildPlanByCommit.CommitID: Commit is nil")
+	}
+
+	if IsErrorResponse(b.Commit.Type) {
+		return "", ProcessCommitError(b.Commit, "Could not get commit ID from commit response")
+	}
+
+	return b.Commit.CommitID, nil
+}
+
+func IsErrorResponse(errorType string) bool {
+	return errorType == ErrorType ||
+		errorType == AlreadyExistsErrorType ||
+		errorType == NoChangeSinceLastCommitErrorType ||
+		errorType == HeadOnBranchMovedErrorType ||
+		errorType == ForbiddenErrorType ||
+		errorType == RemediableSolveErrorType ||
+		errorType == PlanningErrorType
+}
+
+func ProcessCommitError(commit *Commit, fallbackMessage string) error {
+	if commit.Error == nil {
+		return errs.New(fallbackMessage)
+	}
+
+	switch commit.Type {
+	case NotFoundErrorType:
+		return locale.NewInputError("err_buildplanner_commit_not_found", "Could not find commit, recieved message: {{.V0}}", commit.Message)
+	case ParseErrorType:
+		return locale.NewInputError("err_buildplanner_parse_error", "The platform failed to parse the build expression, received the following message: {{.V0}}. Path: {{.V1}}", commit.Message, commit.ParseError.Path)
+	case ForbiddenErrorType:
+		return locale.NewInputError("err_buildplanner_forbidden", "Unable to complete the operation: {{.V0}}, recieved message: {{.V1}}", commit.Operation, commit.Message)
+	case HeadOnBranchMovedErrorType:
+		return locale.NewInputError("err_buildplanner_head_on_branch_moved", "Head on branch has moved, recieved message: {{.V0}}", commit.Error.Message)
+	case NoChangeSinceLastCommitErrorType:
+		return locale.NewInputError("err_buildplanner_no_change_since_last_commit", "No change since last commit, recieved message: {{.V0}}", commit.Error.Message)
+	default:
+		return errs.New(fallbackMessage)
+	}
+}
+
+func ProcessBuildError(build *Build, fallbackMessage string) error {
```

Comment:
The other Process functions do a nil check first, this one doesn't. Is that intentional?